src/MDMTransaction.h

```diff
@@ -27,9 +27,36 @@ NS_SWIFT_NAME(Transaction)
 
 #pragma mark Adding plans to a transaction
 
-/** Associate an plan with a given target. */
+/**
+Associates a plan with a given target.
+
+@param plan The plan to add to this transaction.
+@param target The target on which the plan can operate.
+*/
+- (void)addPlan:(nonnull id<MDMPlan>)plan
+       toTarget:(nonnull id)target
+  NS_SWIFT_NAME(add(plan:to:));
+
+/**
+ Associates a named plan with a given target.
+ 
+ @param plan The plan to add to this transaction.
+ @param target The target on which the plan can operate.
+ @param name String identifier for the plan.
+ */
 - (void)addPlan:(nonnull id<MDMPlan>)plan
        toTarget:(nonnull id)target
-    NS_SWIFT_NAME(add(plan:to:));
+       withName:(nonnull NSString *)name
+  NS_SWIFT_NAME(add(plan:to:withName:));
+
+/**
+ Removes a named plan from the transaction.
+ 
+ @param plan The plan to add to this transaction.
+ @param target The target on which the plan can operate.
+ @param name String identifier for the plan
+ */
+- (void)removePlanNamed:(nonnull NSString *)name
+  NS_SWIFT_NAME(remove(name:));
```

Comment:
Added a `MDMNamedPlanPerforming` protocol which looks like:
```
- (void)addPlan:(nonnull id<MDMPlan>)plan
       withName:(nonnull NSString *)name
  NS_SWIFT_NAME(add(plan:withName:));

- (void)removePlan:(nonnull id<MDMPlan>)plan
          withName:(nonnull NSString *)name
  NS_SWIFT_NAME(remove(plan:withName:));
```
and added the callbacks into `MDMTransaction`.

This change did result in one test failure due to overriding the `isEqual:` method on `MDMTransactionLog` which was needed due to changing the `_namedPlans` to being changed over to `NSMutableSet`.

Can make some changes to fix that test, but wanted to know if this is the correct route to go down.